controllers/chat/ChatController.js

```diff
@@ -181,12 +181,13 @@ module.exports = {
       });
       return res.status(200).json(responseSuccess('Success retrieve channels', channels));
     } catch (error) {
-      await client.disconnectUser();
       return res.status(error.statusCode ?? error.status ?? 400).json({
         status: 'error',
         code: error.statusCode ?? error.status ?? 400,
         message: error.message
       });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling here could be improved by providing more specific error messages based on the type of error that occurred. This would make it easier to debug issues when they arise. Also, consider logging the error for further analysis.

```diff
- return res.status(error.statusCode ?? error.status ?? 400).json({
-   status: 'error',
-   code: error.statusCode ?? error.status ?? 400,
-   message: error.message
- });
+ let statusCode = error.statusCode ?? error.status ?? 400;
+ let errorMessage = error.message;
+ if (statusCode === 500) {
+   errorMessage = 'Internal server error';
+ }
+ console.error(error); // log the error
+ return res.status(statusCode).json({
+   status: 'error',
+   code: statusCode,
+   message: errorMessage
+ });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->